tests/integration/client_relations/test_internal_client_relations.py

```diff
@@ -0,0 +1,146 @@
+    """Tests that when a client is integrated with mongos a user it receives connection info."""
+    await ops_test.model.integrate(APPLICATION_APP_NAME, MONGOS_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[APPLICATION_APP_NAME, MONGOS_APP_NAME], status="active", idle_period=20
+    )
+
+    await ops_test.model.block_until(
+        lambda: is_relation_joined(
+            ops_test,
+            CLIENT_RELATION_NAME,
+            MONGOS_RELATION_NAME,
+        )
+        is True,
+        timeout=600,
+    )
+
+    connection_string = await get_client_connection_string(
+        ops_test, APPLICATION_APP_NAME, CLIENT_RELATION_NAME
+    )
+    assert connection_string, "Connection string  not provided to client."
+
+    username, password = await get_mongos_user_password(
+        ops_test, app_name=APPLICATION_APP_NAME, relation_name="mongos"
+    )
+    assert username, "Username not provided to client."
+    assert password, "Password not provided to client."
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_user_can_connect(ops_test: OpsTest) -> None:
+    """Tests that the user created by mongos can connect with auth."""
+    username, password = await get_mongos_user_password(
+        ops_test, app_name=APPLICATION_APP_NAME, relation_name="mongos"
+    )
+    assert username, "Username not provided to client"
+    assert password, "Password not provided to client"
+
+    mongos_host = await get_address_of_unit(ops_test, unit_id=0)
+    client_user_uri = f"mongodb://{username}:{password}@{mongos_host}:{MONGOS_PORT}"
+    mongos_can_connect_with_auth = await check_mongos(ops_test, uri=client_user_uri)
+    assert mongos_can_connect_with_auth, "User created cannot connect with auth."
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_user_with_extra_roles(ops_test: OpsTest) -> None:
+    """Tests that the user created by mongos router has the permissions it asks for."""
+    username, password = await get_mongos_user_password(
+        ops_test, app_name=APPLICATION_APP_NAME, relation_name="mongos"
+    )
+    mongos_host = await get_address_of_unit(ops_test, unit_id=0)
+    client_user_uri = f"mongodb://{username}:{password}@{mongos_host}:{MONGOS_PORT}"
+
+    mongos_client = await get_direct_mongos_client(ops_test, uri=client_user_uri)
+    mongos_client.admin.command(
+        "createUser",
+        TEST_USER_NAME,
+        pwd=TEST_USER_PWD,
+        roles=[{"role": "readWrite", "db": TEST_DB_NAME}],
+        mechanisms=["SCRAM-SHA-256"],
+    )
+    mongos_client.close()
+
+    mongos_host = await get_address_of_unit(ops_test, unit_id=0)
+    test_user_uri = (
+        f"mongodb://{TEST_USER_NAME}:{TEST_USER_PWD}@{mongos_host}:{MONGOS_PORT}"
+    )
+    test_user_accessible = await check_mongos(ops_test, uri=test_user_uri)
+    assert test_user_accessible, "User created is not accessible."
+
+
+@pytest.mark.group(1)
```

Comment:
I would:
```suggestion
@pytest.mark.group(1)
@pytest.mark.abort_on_fail
```
This way when we add more tests, we won't have issues with that.